coalib/files/Fileproxy.py

```diff
@@ -0,0 +1,97 @@
+from coala_utils.decorators import generate_eq
+
+
+@generate_eq("filename")
+class Fileproxy:
+    """
+    The file proxy is a wrapper object to wrap every file. The file proxy is an
+    object that contains properties about the file currently processed (such as
+    the filename). The fileproxy object also protects your file from being
```

Comment:
"protects your file" - eh?  what is being protected?

A bear can still `open()` the file and write to it.

A bear can still edit the contents of the fileproxy object using `fp[0][:] = 'abc'`.